Get rid of `location.siblings` call

esoergel opened the pull request:
@proteusvacuum

esoergel committed:
Get rid of `siblings` call

esoergel commented:
Oh whoops, @benrudolph is my code buddy

benrudolph commented on corehq/apps/locations/forms.py:
```diff
@@ -195,10 +195,14 @@ def clean_parent_id(self):
+            if (SQLLocation.objects.filter(domain=self.location.domain,
+                                           parent=parent,
+                                           name=name)
+                                   .exclude(location_id=self.location._id)
```
does this mean we have locations that reference themselves as parents? haha

esoergel replied:
I certainly hope not!
This is getting all locations which list `parent` as the `parent`, but excludes the current location.  The idea is, when setting the parent, make sure that there are no other children of that parent with the same name.

benrudolph commented:
👍 on ✔️

esoergel committed:
children > get_children() for consistency with SQL

snopoke commented:
nice to see some locations cleanup happening!

proteusvacuum commented:
kicked build

benrudolph commented:
```
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tests/test_report_reminders.py", line 136, in test_basic_report_no_responses
    result = construct_randr_summary(self.district)
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tanzania/reminders/reports.py", line 61, in construct_randr_summary
    get_business_day_of_month_before(now.year, now.month, 5)
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tanzania/reminders/reports.py", line 50, in construct_summary
    ret["total"] = len(locations)
TypeError: object of type 'itertools.imap' has no len()
```
could be related to removing `list` from the output of `get_children` though didn't look very hard

esoergel committed:
Coerce to a list for backwards compatability

I'm not just making `Location.get_children()` return a list, since the
goal is to be able to make it api compatible with
`SQLLocation.get_children()`

esoergel committed:
Clean up tests

esoergel committed:
Handle couch parent locations

esoergel commented:
:white_check_mark:

snopoke commented:
🎈 Nice
